opm/simulators/flow/BlackoilModel.hpp

```diff
@@ -591,19 +590,18 @@ namespace Opm {
                     }
                     for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++ phaseIdx) {
                         Scalar tmpSat = saturationsNew[phaseIdx] - saturationsOld[phaseIdx];
-                        resultDelta += tmpSat*tmpSat;
-                        resultDenom += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
-                        assert(std::isfinite(resultDelta));
-                        assert(std::isfinite(resultDenom));
+                        resultDeltaSaturation += tmpSat*tmpSat;
+                        resultDenomSaturation += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
+                        assert(std::isfinite(resultDeltaSaturation));
+                        assert(std::isfinite(resultDenomSaturation));
                     }
                 }
             }
 
-            resultDelta = gridView.comm().sum(resultDelta);
-            resultDenom = gridView.comm().sum(resultDenom);
+            gridView.comm.sum(r.data(), 4);
```

Comment:
Slight generalisation: The hard-coded value `4` here is better written as `r.size()` instead.

Reply:
Quick follow-up: `gridView.comm.sum()` should be `gridView.comm().sum()` instead.

Reply:
Done!